### PR TITLE
Issue/547/improve webpack configuration

### LINKED ISSUE #547: Improve webpack configuration

### PR DESCRIPTION
**Pull request by dbetka**
close #547 

What done:
- Add progress bar to webpack console output
- Create AppConsoleFramePlugin for webpack
- Split webpack.common.js to plugins and utils
- Show mode, target and is-watching in webpack console output
- Change structure of webpack files
- Add dependency sizes analyzer
- Add assets sizes list into AppConsoleFramePlugin
- Extract guide modules to separate async file
- Extract ellipse-progress modules to separate async file
- Extract pages after login into separate async file
- Improve webpack environment variables
- Fix i18n TS type

### CODE REVIEW

File: client/webpack/plugins/app-console-frame.js [hunk 1]
```diff
@@ -0,0 +1,148 @@
+const chalk = require('chalk');
+const dayjs = require('dayjs');
```
Comment (Bedpaw) [inline client/webpack/plugins/app-console-frame.js:2]: dayjs shouldn't be used directly

File: client/webpack/plugins/app-console-frame.js [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+            if (compiler.options.watch) {
+              this.write('  Waiting for changes...');
+            }
+          }, 100);
+        });
+      },
+    );
+  }
+
+  capitalizeFirstChar (string) {
```
Comment (Bedpaw) [inline client/webpack/plugins/app-console-frame.js:61]: could be as utils/string

File: client/webpack/plugins/app-console-frame.js (hunk 1 above)
Comment (Bedpaw) [reply client/webpack/plugins/app-console-frame.js:2]: In general this file is complicated and could be refactored

File: client/webpack/plugins/progress-bar-config.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
+const ProgressBarWebpackPlugin = require('progress-bar-webpack-plugin');
+const chalk = require('chalk');
+
+class ProgressBarConfig {
+  constructor () {
+    return new ProgressBarWebpackPlugin({
+      format: '  Build ' + chalk.bgGray(':bar') + ' ' + chalk.green.bold(':percent') + ' ',
```
Comment (Bedpaw) [inline client/webpack/plugins/progress-bar-config.js:7]: You should use template string if possible

File: client/webpack/plugins/app-console-frame.js (hunk 1 above)
Comment (dbetka) [reply client/webpack/plugins/app-console-frame.js:2]: Configuration files should separate from frontend files. Day stay used directly.

File: client/webpack/plugins/app-console-frame.js (hunk 2 above)
Comment (dbetka) [reply client/webpack/plugins/app-console-frame.js:61]: Configuration files should separate from front-end files.

File: client/webpack/plugins/progress-bar-config.js (hunk 3 above)
Comment (dbetka) [reply client/webpack/plugins/progress-bar-config.js]: Done

**Pull request merged**